### ISSUE TITLE
Difference in storage engine performance for Neo on cold data

### ISSUE THREAD

**Issue by nemanja-boric-sociomantic** [sociomantic-tsunami/dlsnode#31]
During the recent deployment of test application we've noticed the slowdown when using Neo protocol which we managed to pin down to the apparent difference in the node's performance when the page cache is empty. When the requested data is already paged in in the memory, the difference matches what we were seeing when we did dry run client tests.

- The results with the data paged in:

Legacy:

```
Timing ---
Summary: Reading time:                                 60.85
Summary: Total dls time:                               60.85
Summary: Sorting time:                                 0.00
Summary: Reduction time:                               8.35
Summary: Writing time:                                 0.00
Summary: Copying time:                                 0.00
Summary: Total time:                                   69.20
```

Neo:

```
Timing ---
Summary: Reading time:                                 64.93
Summary: Total dls time:                               64.93
Summary: Sorting time:                                 0.00
Summary: Reduction time:                               8.13
Summary: Writing time:                                 0.00
Summary: Copying time:                                 0.00
Summary: Total time:                                   73.06
```

- The results with the data not in cache:

Legacy:

```Timing ---
Summary: Reading time:                                 62.08
Summary: Total dls time:                               62.08
Summary: Sorting time:                                 0.00
Summary: Reduction time:                               8.49
Summary: Writing time:                                 0.00
Summary: Copying time:                                 0.00
Summary: Total time:                                   70.57
```

Neo:

```
Timing ---
Summary: Reading time:                                 81.66
Summary: Total dls time:                               81.66
Summary: Sorting time:                                 0.00
Summary: Reduction time:                               8.42
Summary: Writing time:                                 0.00
Summary: Copying time:                                 0.00
Summary: Total time:                                   90.09
```

The results are consistent across multiple testing rounds.

This shows that there's no unexpected difference in the Neo _protocol_ overhead. When serving the data that's in, Neo is capable of achieving the approximately same performance. This is good news, since we did all the tricks that we could think of in the _swarm_ and _dlsproto_'s implementation. The only thing that we never tested/nor optimized is the dlsnode's storage engine neo path, which differs from the legacy one.

**Comment by nemanja-boric-sociomantic** [comment]
The reason is that the Neo Dls Node opens the bucket file and reads the bucket header in the main thread (using blocking API). This means that as soon as there are more than one client open, the performance will suffer.

This is the place where it's done (see how it passes `null` for the `JobNotification`):

https://github.com/sociomantic-tsunami/dlsnode/blob/433771caf01f3d7421508e7e992c24aa17f9049a/src/dlsnode/storage/iterator/NeoStorageEngineStepIterator.d#L416-L417****

**Comment by nemanja-boric-sociomantic** [comment]
Fixed in neo-beta-1 (#35)
